Intrepid2: Use of std::is_pod is deprecated in C++ 20

@trilinos/intrepid2 

## Enhancement
I tried a build with C++ 20 and got a bunch of warnings:
```
packages/intrepid2/src/Shared/Intrepid2_Utils.hpp: In function 'constexpr int Intrepid2::getVectorSizeForHierarchicalParallelism()':
packages/intrepid2/src/Shared/Intrepid2_Utils.hpp:827:17: warning: 'template<class _Tp> struct std::is_pod' is deprecated: use is_standard_layout && is_trivial instead [-Wdeprecated-declarations]
```

--- a/packages/intrepid2/src/Cell/Intrepid2_CellData.hpp
+++ b/packages/intrepid2/src/Cell/Intrepid2_CellData.hpp
@@ -337,10 +337,10 @@ template<unsigned CellTopologyKey>
   */
   template<>
   struct PointInclusion<shards::Line<>::key> {
-    template<typename PointViewType>
+    template<typename PointViewType, typename ScalarType>
     KOKKOS_INLINE_FUNCTION
     static bool
-    check(const PointViewType &point, const double threshold);   
+    check(const PointViewType &point, const ScalarType threshold);   
   };
   
   /** 
@@ -348,10 +348,10 @@ template<unsigned CellTopologyKey>
   */
   template<>
   struct PointInclusion<shards::Triangle<>::key> {
-    template<typename PointViewType>
+    template<typename PointViewType, typename ScalarType>
     KOKKOS_INLINE_FUNCTION
     static bool
-    check(const PointViewType &point, const double threshold);
+    check(const PointViewType &point, const ScalarType threshold);
   };
   
   /** 
@@ -360,10 +360,10 @@ template<unsigned CellTopologyKey>
   template<>
   struct PointInclusion<shards::Quadrilateral<>::key> {
 
-    template<typename PointViewType>
+    template<typename PointViewType, typename ScalarType>
     KOKKOS_INLINE_FUNCTION
     static bool
-    check(const PointViewType &point, const double threshold);
+    check(const PointViewType &point, const ScalarType threshold);
   };
     
   /** 
@@ -371,10 +371,10 @@ template<unsigned CellTopologyKey>
   */
   template<>
   struct PointInclusion<shards::Tetrahedron<>::key> {
-    template<typename PointViewType>
+    template<typename PointViewType, typename ScalarType>
     KOKKOS_INLINE_FUNCTION
     static bool
-    check(const PointViewType &point, const double threshold);
+    check(const PointViewType &point, const ScalarType threshold);
   };
 
   /** 
@@ -382,10 +382,10 @@ template<unsigned CellTopologyKey>
   */
   template<>
   struct PointInclusion<shards::Hexahedron<>::key> {
-    template<typename PointViewType>
+    template<typename PointViewType, typename ScalarType>
     KOKKOS_INLINE_FUNCTION
     static bool
-    check(const PointViewType &point, const double threshold);
+    check(const PointViewType &point, const ScalarType threshold);
   };
   
   /** 
@@ -393,10 +393,10 @@ template<unsigned CellTopologyKey>
   */
   template<>
   struct PointInclusion<shards::Pyramid<>::key> {
-    template<typename PointViewType>
+    template<typename PointViewType, typename ScalarType>
     KOKKOS_INLINE_FUNCTION
     static bool
-    check(const PointViewType &point, const double threshold);
+    check(const PointViewType &point, const ScalarType threshold);
   };
 
   /** 
@@ -404,10 +404,10 @@ template<unsigned CellTopologyKey>
   */
   template<>
   struct PointInclusion<shards::Wedge<>::key> {
-    template<typename PointViewType>
+    template<typename PointViewType, typename ScalarType>
     KOKKOS_INLINE_FUNCTION
     static bool
-    check(const PointViewType &point, const double threshold);
+    check(const PointViewType &point, const ScalarType threshold);
   };
 
 }

--- a/packages/intrepid2/src/Cell/Intrepid2_CellDataDef.hpp
+++ b/packages/intrepid2/src/Cell/Intrepid2_CellDataDef.hpp
@@ -831,7 +831,7 @@ refCenterDataStatic_ = {
   bool
   PointInclusion<shards::Line<>::key>::
   check(const PointViewType &point, const ScalarType threshold) {
-    const double minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
+    const ScalarType minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
     return (minus_one <= point(0) && point(0) <= plus_one);
   }  
 
@@ -840,7 +840,7 @@ refCenterDataStatic_ = {
   bool
   PointInclusion<shards::Triangle<>::key>::
   check(const PointViewType &point, const ScalarType threshold) {
-    const double distance = max( max( -point(0), -point(1) ), point(0) + point(1) - 1.0 );
+    const ScalarType distance = max( max( -point(0), -point(1) ), point(0) + point(1) - 1.0 );
     return distance < threshold;
   }
   
@@ -850,7 +850,7 @@ refCenterDataStatic_ = {
   PointInclusion<shards::Quadrilateral<>::key>::
   check(const PointViewType &point, 
                       const ScalarType threshold) {
-    const double minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
+    const ScalarType minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
     return ((minus_one <= point(0) && point(0) <= plus_one) &&
             (minus_one <= point(1) && point(1) <= plus_one));
   }  
@@ -860,7 +860,7 @@ refCenterDataStatic_ = {
   bool
   PointInclusion<shards::Tetrahedron<>::key>::
   check(const PointViewType &point, const ScalarType threshold) {
-    const double distance = max( max(-point(0),-point(1)),
+    const ScalarType distance = max( max(-point(0),-point(1)),
                                   max(-point(2), point(0) + point(1) + point(2) - 1) );
     return distance < threshold;
   }
@@ -870,7 +870,7 @@ refCenterDataStatic_ = {
   bool
   PointInclusion<shards::Hexahedron<>::key>::
   check(const PointViewType &point, const ScalarType threshold) {
-    const double minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
+    const ScalarType minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
     return ((minus_one <= point(0) && point(0) <= plus_one) &&
             (minus_one <= point(1) && point(1) <= plus_one) &&
             (minus_one <= point(2) && point(2) <= plus_one));
@@ -881,9 +881,9 @@ refCenterDataStatic_ = {
   bool
   PointInclusion<shards::Pyramid<>::key>::
   check(const PointViewType &point, const ScalarType threshold) {
-    const double minus_one = -1.0 - threshold, plus_one = 1.0 + threshold, minus_zero = -threshold;
-    const double left  = minus_one + point(2);
-    const double right =  plus_one - point(2);
+    const ScalarType minus_one = -1.0 - threshold, plus_one = 1.0 + threshold, minus_zero = -threshold;
+    const ScalarType left  = minus_one + point(2);
+    const ScalarType right =  plus_one - point(2);
     return ((left       <= point(0) && point(0) <= right) &&
             (left       <= point(1) && point(1) <= right) &&
             (minus_zero <= point(2) && point(2) <= plus_one));
@@ -894,8 +894,8 @@ refCenterDataStatic_ = {
   bool
   PointInclusion<shards::Wedge<>::key>::
   check(const PointViewType &point, const ScalarType threshold) {
-    const double minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
-    const double distance = max( max( -point(0), -point(1) ), point(0) + point(1) - 1 );
+    const ScalarType minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
+    const ScalarType distance = max( max( -point(0), -point(1) ), point(0) + point(1) - 1 );
     return (distance < threshold && (minus_one <= point(2) && point(2) <= plus_one));
   }
 

--- a/packages/intrepid2/src/Cell/Intrepid2_CellDataDef.hpp
+++ b/packages/intrepid2/src/Cell/Intrepid2_CellDataDef.hpp
@@ -826,61 +826,61 @@ refCenterDataStatic_ = {
 // Point Inclusion 
 
 
-  template<typename PointViewType>
+  template<typename PointViewType, typename ScalarType>
   KOKKOS_INLINE_FUNCTION
   bool
   PointInclusion<shards::Line<>::key>::
-  check(const PointViewType &point, const double threshold) {
+  check(const PointViewType &point, const ScalarType threshold) {
     const double minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
     return (minus_one <= point(0) && point(0) <= plus_one);
   }  
 
-  template<typename PointViewType>
+  template<typename PointViewType, typename ScalarType>
   KOKKOS_INLINE_FUNCTION
   bool
   PointInclusion<shards::Triangle<>::key>::
-  check(const PointViewType &point, const double threshold) {
+  check(const PointViewType &point, const ScalarType threshold) {
     const double distance = max( max( -point(0), -point(1) ), point(0) + point(1) - 1.0 );
     return distance < threshold;
   }
   
-  template<typename PointViewType>
+  template<typename PointViewType, typename ScalarType>
   KOKKOS_INLINE_FUNCTION
   bool
   PointInclusion<shards::Quadrilateral<>::key>::
   check(const PointViewType &point, 
-                      const double threshold) {
+                      const ScalarType threshold) {
     const double minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
     return ((minus_one <= point(0) && point(0) <= plus_one) &&
             (minus_one <= point(1) && point(1) <= plus_one));
   }  
 
-  template<typename PointViewType>
+  template<typename PointViewType, typename ScalarType>
   KOKKOS_INLINE_FUNCTION
   bool
   PointInclusion<shards::Tetrahedron<>::key>::
-  check(const PointViewType &point, const double threshold) {
+  check(const PointViewType &point, const ScalarType threshold) {
     const double distance = max( max(-point(0),-point(1)),
                                   max(-point(2), point(0) + point(1) + point(2) - 1) );
     return distance < threshold;
   }
 
-  template<typename PointViewType>
+  template<typename PointViewType, typename ScalarType>
   KOKKOS_INLINE_FUNCTION
   bool
   PointInclusion<shards::Hexahedron<>::key>::
-  check(const PointViewType &point, const double threshold) {
+  check(const PointViewType &point, const ScalarType threshold) {
     const double minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
     return ((minus_one <= point(0) && point(0) <= plus_one) &&
             (minus_one <= point(1) && point(1) <= plus_one) &&
             (minus_one <= point(2) && point(2) <= plus_one));
   }
   
-  template<typename PointViewType>
+  template<typename PointViewType, typename ScalarType>
   KOKKOS_INLINE_FUNCTION
   bool
   PointInclusion<shards::Pyramid<>::key>::
-  check(const PointViewType &point, const double threshold) {
+  check(const PointViewType &point, const ScalarType threshold) {
     const double minus_one = -1.0 - threshold, plus_one = 1.0 + threshold, minus_zero = -threshold;
     const double left  = minus_one + point(2);
     const double right =  plus_one - point(2);
@@ -889,11 +889,11 @@ refCenterDataStatic_ = {
             (minus_zero <= point(2) && point(2) <= plus_one));
   }
 
-  template<typename PointViewType>
+  template<typename PointViewType, typename ScalarType>
   KOKKOS_INLINE_FUNCTION
   bool
   PointInclusion<shards::Wedge<>::key>::
-  check(const PointViewType &point, const double threshold) {
+  check(const PointViewType &point, const ScalarType threshold) {
     const double minus_one = -1.0 - threshold, plus_one = 1.0 + threshold;
     const double distance = max( max( -point(0), -point(1) ), point(0) + point(1) - 1 );
     return (distance < threshold && (minus_one <= point(2) && point(2) <= plus_one));

--- a/packages/intrepid2/src/Cell/Intrepid2_CellTools.hpp
+++ b/packages/intrepid2/src/Cell/Intrepid2_CellTools.hpp
@@ -1396,11 +1396,13 @@ public:
         \param  threshold         [in]  - "tightness" of the inclusion test
         \return true if the point is in the closure of the specified reference cell and false otherwise.
     */
-    template<typename pointViewType>
+    template<typename PointViewType>
     static bool 
-    checkPointInclusion( const pointViewType        point,
+    checkPointInclusion( const PointViewType        point,
                          const shards::CellTopology cellTopo,
-                         const double               thres = threshold() );
+                         const typename ScalarTraits<typename PointViewType::value_type>::scalar_type thres = 
+                               threshold<typename ScalarTraits<typename PointViewType::value_type>::scalar_type>() );
+
 
 
     /** \brief  Checks every point for inclusion in the reference cell of a given topology.
@@ -1417,7 +1419,8 @@ public:
              typename InputViewType>
     static void checkPointwiseInclusion(       OutputViewType inCell, 
                                          const InputViewType points,
-                                         const double thresh = threshold()); 
+                                         const typename ScalarTraits<typename InputViewType::value_type>::scalar_type thresh =
+                                               threshold<typename ScalarTraits<typename InputViewType::value_type>::scalar_type>()); 
 
 
 
@@ -1434,7 +1437,8 @@ public:
     static void checkPointwiseInclusion(       InCellViewType inCell,                     
                                          const PointViewType points,                       
                                          const shards::CellTopology cellTopo,                                                       
-                                         const double thres = threshold() );
+                                         const typename ScalarTraits<typename PointViewType::value_type>::scalar_type thres = 
+                                               threshold<typename ScalarTraits<typename PointViewType::value_type>::scalar_type>() );
 
     /** \brief  Checks every points for inclusion in physical cells from a cell workset.
                 The points can belong to a global set and stored in a rank-2 (P,D) view,
@@ -1454,7 +1458,8 @@ public:
                                          const Kokkos::DynRankView<pointValueType,pointProperties...> points,                       
                                          const Kokkos::DynRankView<cellWorksetValueType,cellWorksetProperties...> cellWorkset,      
                                          const shards::CellTopology cellTopo,                                                       
-                                         const double thres = threshold() );
+                                         const typename ScalarTraits<pointValueType>::scalar_type thres = 
+                                               threshold<typename ScalarTraits<pointValueType>::scalar_type>() );
 
 
     // //============================================================================================//

--- a/packages/intrepid2/src/Cell/Intrepid2_CellToolsDefInclusion.hpp
+++ b/packages/intrepid2/src/Cell/Intrepid2_CellToolsDefInclusion.hpp
@@ -34,9 +34,9 @@ namespace Intrepid2 {
   template<typename PointViewType>
   bool 
   CellTools<DeviceType>::
-  checkPointInclusion( const PointViewType        point,
-                       const shards::CellTopology cellTopo,
-                       const double               threshold) {
+  checkPointInclusion( const PointViewType          point,
+                       const shards::CellTopology   cellTopo,
+                       const typename ScalarTraits<typename PointViewType::value_type>::scalar_type threshold) {
 #ifdef HAVE_INTREPID2_DEBUG
     INTREPID2_TEST_FOR_EXCEPTION( point.rank() != 1, std::invalid_argument,
                                   ">>> ERROR (Intrepid2::CellTools::checkPointInclusion): Point must have rank 1. ");
@@ -94,12 +94,13 @@ namespace Intrepid2 {
   struct checkPointInclusionFunctor {
     OutputViewType output_;
     InputViewType input_;
-    double threshold_;
+    using ScalarType = typename ScalarTraits<typename InputViewType::value_type>::scalar_type;
+    ScalarType threshold_;
 
     KOKKOS_INLINE_FUNCTION
-    checkPointInclusionFunctor(       OutputViewType output,
-                               const  InputViewType  input,
-                               const  double         threshold)
+    checkPointInclusionFunctor(       OutputViewType                      output,
+                               const  InputViewType                       input,
+                               const  ScalarType threshold)
       : output_(output), 
         input_(input),
         threshold_(threshold) {}
@@ -129,7 +130,7 @@ namespace Intrepid2 {
   void CellTools<DeviceType>::
   checkPointwiseInclusion(      OutputViewType inCell, 
                           const InputViewType  points,
-                          const double         threshold) {     
+                          const typename ScalarTraits<typename InputViewType::value_type>::scalar_type threshold) {     
 
      using FunctorType = checkPointInclusionFunctor<cellTopologyKey,decltype(inCell),decltype(points)>;
     if (points.rank() == 2) {     // inCell.rank() == 1
@@ -144,13 +145,13 @@ namespace Intrepid2 {
 
   template<typename DeviceType>
   template<typename InCellViewType,
-           typename PointViewType>
+           typename InputViewType>
   void
   CellTools<DeviceType>::
-  checkPointwiseInclusion(       InCellViewType       inCell,
-                           const PointViewType        points,
-                           const shards::CellTopology cellTopo,
-                           const double               threshold ) {
+  checkPointwiseInclusion(       InCellViewType         inCell,
+                           const InputViewType          points,
+                           const shards::CellTopology   cellTopo,
+                           const typename ScalarTraits<typename InputViewType::value_type>::scalar_type threshold ) {
 #ifdef HAVE_INTREPID2_DEBUG
     {
       INTREPID2_TEST_FOR_EXCEPTION( (inCell.rank() != 1) && (inCell.rank() != 2), std::invalid_argument,
@@ -218,7 +219,7 @@ namespace Intrepid2 {
                            const Kokkos::DynRankView<pointValueType,pointProperties...> points,
                            const Kokkos::DynRankView<cellWorksetValueType,cellWorksetProperties...> cellWorkset,
                            const shards::CellTopology cellTopo,
-                           const double threshold ) {
+                           const typename ScalarTraits<pointValueType>::scalar_type threshold ) {
 #ifdef HAVE_INTREPID2_DEBUG
     {
       const auto key = cellTopo.getBaseKey();

--- a/packages/intrepid2/src/Discretization/Integration/Intrepid2_IntegrationToolsDef.hpp
+++ b/packages/intrepid2/src/Discretization/Integration/Intrepid2_IntegrationToolsDef.hpp
@@ -123,7 +123,7 @@ namespace Intrepid2 {
         // prepare for allocation of temporary storage
         // note: tempStorage goes "backward", starting from the final component, which needs just one entry
         
-        const bool allocateFadStorage = !std::is_pod<Scalar>::value;
+        const bool allocateFadStorage = !(std::is_standard_layout<Scalar>::value && std::is_trivial<Scalar>::value);  
         if (allocateFadStorage)
         {
           fad_size_output_ = dimension_scalar(integralView_);
@@ -1063,7 +1063,7 @@ namespace Intrepid2 {
         // prepare for allocation of temporary storage
         // note: tempStorage goes "backward", starting from the final component, which needs just one entry
 
-        const bool allocateFadStorage = !std::is_pod<Scalar>::value;
+        const bool allocateFadStorage = !(std::is_standard_layout<Scalar>::value && std::is_trivial<Scalar>::value);
         if (allocateFadStorage)
         {
           fad_size_output_ = dimension_scalar(integralView_);
@@ -2127,7 +2127,7 @@ void IntegrationTools<DeviceType>::integrate(Data<Scalar,DeviceType> integrals, 
               {
                 ScalarView<Scalar,DeviceType> componentIntegralView;
                 
-                const bool allocateFadStorage = !std::is_pod<Scalar>::value;
+                const bool allocateFadStorage = !(std::is_standard_layout<Scalar>::value && std::is_trivial<Scalar>::value);
                 if (allocateFadStorage)
                 {
                   auto fad_size_output = dimension_scalar(integrals.getUnderlyingView());

--- a/packages/intrepid2/src/Shared/Intrepid2_TestUtils.hpp
+++ b/packages/intrepid2/src/Shared/Intrepid2_TestUtils.hpp
@@ -203,7 +203,7 @@ namespace Intrepid2
   template<typename ValueType, typename DeviceType, class ... DimArgs>
   inline ViewType<ValueType,DeviceType> getView(const std::string &label, DimArgs... dims)
   {
-    const bool allocateFadStorage = !std::is_pod<ValueType>::value;
+    const bool allocateFadStorage = !(std::is_standard_layout<ValueType>::value && std::is_trivial<ValueType>::value);
     if (!allocateFadStorage)
     {
       return ViewType<ValueType,DeviceType>(label,dims...);
@@ -218,7 +218,7 @@ namespace Intrepid2
   template<typename ValueType, class ... DimArgs>
   inline FixedRankViewType< typename RankExpander<ValueType, sizeof...(DimArgs) >::value_type, DefaultTestDeviceType > getFixedRankView(const std::string &label, DimArgs... dims)
   {
-    const bool allocateFadStorage = !std::is_pod<ValueType>::value;
+    const bool allocateFadStorage = !(std::is_standard_layout<ValueType>::value && std::is_trivial<ValueType>::value);
     using value_type = typename RankExpander<ValueType, sizeof...(dims) >::value_type;
     if (!allocateFadStorage)
     {

--- a/packages/intrepid2/src/Shared/Intrepid2_Types.hpp
+++ b/packages/intrepid2/src/Shared/Intrepid2_Types.hpp
@@ -65,14 +65,26 @@ namespace Intrepid2 {
     return epsilon<double>();
   }
 
+  template<typename ValueType>
+  KOKKOS_FORCEINLINE_FUNCTION
+  ValueType tolerence() {
+    return 100.0*epsilon<ValueType>();
+  }
+
   KOKKOS_FORCEINLINE_FUNCTION
   double tolerence() {
-    return 100.0*epsilon();
+    return tolerence<double>();
+  }
+
+  template<typename ValueType>
+  KOKKOS_FORCEINLINE_FUNCTION
+  ValueType threshold() {
+    return 10.0*epsilon<ValueType>();
   }
 
   KOKKOS_FORCEINLINE_FUNCTION
   double threshold() {
-    return 10.0*epsilon();
+    return threshold<double>();
   }
 
   /// Define constants

--- a/packages/intrepid2/src/Shared/Intrepid2_Utils.hpp
+++ b/packages/intrepid2/src/Shared/Intrepid2_Utils.hpp
@@ -281,13 +281,13 @@ namespace Intrepid2 {
   template<typename T>
   KOKKOS_FORCEINLINE_FUNCTION
   constexpr typename
-  std::enable_if< !std::is_pod<T>::value, typename ScalarTraits<T>::scalar_type >::type
+  std::enable_if< !(std::is_standard_layout<T>::value && std::is_trivial<T>::value), typename ScalarTraits<T>::scalar_type >::type
   get_scalar_value(const T& obj) {return obj.val();}
 
   template<typename T>
   KOKKOS_FORCEINLINE_FUNCTION
   constexpr typename
-  std::enable_if< std::is_pod<T>::value, typename ScalarTraits<T>::scalar_type >::type
+  std::enable_if< std::is_standard_layout<T>::value && std::is_trivial<T>::value, typename ScalarTraits<T>::scalar_type >::type
   get_scalar_value(const T& obj){return obj;}
 
 
@@ -300,13 +300,13 @@ namespace Intrepid2 {
   template<typename T, typename ...P>
   KOKKOS_INLINE_FUNCTION
   constexpr typename
-  std::enable_if< std::is_pod<T>::value, unsigned >::type
+  std::enable_if< std::is_standard_layout<T>::value && std::is_trivial<T>::value, unsigned >::type
   dimension_scalar(const Kokkos::DynRankView<T, P...> /* view */) {return 1;}
 
   template<typename T, typename ...P>
   KOKKOS_INLINE_FUNCTION
   constexpr typename
-  std::enable_if< std::is_pod< typename Kokkos::View<T, P...>::value_type >::value, unsigned >::type
+  std::enable_if< std::is_standard_layout<typename Kokkos::View<T, P...>::value_type>::value && std::is_trivial<typename Kokkos::View<T, P...>::value_type>::value, unsigned >::type
   dimension_scalar(const Kokkos::View<T, P...> /*view*/) {return 1;}
 
   template<typename T, typename ...P>
@@ -339,7 +339,7 @@ namespace Intrepid2 {
     using DeviceType         = typename ViewType::device_type;
     using ViewTypeWithLayout = Kokkos::DynRankView<ValueType, ResultLayout, DeviceType >;
     
-    const bool allocateFadStorage = !std::is_pod<ValueType>::value;
+    const bool allocateFadStorage = !(std::is_standard_layout<ValueType>::value && std::is_trivial<ValueType>::value);
     if (!allocateFadStorage)
     {
       return ViewTypeWithLayout(label,dims...);
@@ -766,7 +766,7 @@ namespace Intrepid2 {
   template <typename ValueType>
   struct NaturalLayoutForType {
     using layout  =
-    typename std::conditional<std::is_pod<ValueType>::value,
+    typename std::conditional<(std::is_standard_layout<ValueType>::value && std::is_trivial<ValueType>::value),
       Kokkos::LayoutLeft, // for POD types, use LayoutLeft
       Kokkos::LayoutNatural<Kokkos::LayoutLeft> >::type; // For FAD types, use LayoutNatural
   };
@@ -791,7 +791,7 @@ namespace Intrepid2 {
   template<typename Scalar>
   constexpr int getVectorSizeForHierarchicalParallelism()
   {
-    return std::is_pod<Scalar>::value ? VECTOR_SIZE : FAD_VECTOR_SIZE;
+    return (std::is_standard_layout<Scalar>::value && std::is_trivial<Scalar>::value) ? VECTOR_SIZE : FAD_VECTOR_SIZE;
   }
   
   /**
@@ -803,7 +803,7 @@ namespace Intrepid2 {
   KOKKOS_INLINE_FUNCTION
   constexpr unsigned getScalarDimensionForView(const ViewType &view)
   {
-    return (std::is_pod<typename ViewType::value_type>::value) ? 0 : get_dimension_scalar(view);
+    return (std::is_standard_layout<typename ViewType::value_type>::value && std::is_trivial<typename ViewType::value_type>::value) ? 0 : get_dimension_scalar(view);
   }
 } // end namespace Intrepid2
 

--- a/packages/sacado/src/new_design/Sacado_Fad_Exp_ViewStorage.hpp
+++ b/packages/sacado/src/new_design/Sacado_Fad_Exp_ViewStorage.hpp
@@ -79,7 +79,7 @@ namespace Sacado {
       //! Constructor
       SACADO_INLINE_FUNCTION
       ViewStorage(T* v, const int arg_size = 0, const int arg_stride = 0) :
-        sz_(arg_size), stride_(arg_stride), val_(v+sz_.value*stride_.value), dx_(v) {}
+        sz_(arg_size), stride_(arg_stride), val_(v+sz_.value*static_cast<int>(stride_.value)), dx_(v) {}
 
       //! Constructor
       SACADO_INLINE_FUNCTION

